Assets/HoloToolkit/Utilities/Scripts/SceneLauncher.cs

```diff
@@ -1,90 +1,71 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-using HoloToolkit.Unity.InputModule;
+using System;
 using System.Collections.Generic;
 using UnityEngine;
-using UnityEngine.Events;
 using UnityEngine.SceneManagement;
 
 namespace HoloToolkit.Unity
 {
     public class SceneLauncher : MonoBehaviour
     {
+        [SerializeField]
+        [Tooltip("The list of scenes in the build.  Updating this list, also updates the scene build settings!")]
+        public List<string> SceneList;
+
+        [SerializeField]
+        [Tooltip("Location of the center of the grid of buttons in Unity space.")]
+        public GameObject ButtonSpawnLocation;
+
+        [SerializeField]
         [Tooltip("Prefab used as a button for each scene.")]
         public SceneLauncherButton SceneButtonPrefab;
-        [Tooltip("Location of the center of the grid of buttons in Unity space.")]
-        public Vector3 ButtonCenterLocation = new Vector3(0, 0, 1);
+
+        [SerializeField]
         [Tooltip("Number of rows in the grid of buttons. As more scenes are added, they will spread out horizontally using this number of rows.")]
         public int MaxRows = 5;
-        [Tooltip("Prefab that will continue running when another scene is launched, offering a command to return to the scene launcher.")]
-        public KeywordManager ReturnToSceneLauncherPrefab;
 
+        public int SceneLauncherBuildIndex { get; private set; }
         private Vector3 sceneButtonSize = Vector3.one;
 
-        private void OnValidate()
-        {
-            Debug.Assert(SceneButtonPrefab != null, "SceneLauncher.SceneButtonPrefab is not set.");
```

Comment:
I wasn't aware of that. Can you tell me more about warnings about unassigned references? If you don't attach a game object and SceneButtonPrefab is left as null then where would the warning appear?